test/helpers/deploy/deploy_resources.go

```diff
@@ -1010,6 +1010,56 @@ func KongDataPlaneClientCertificateAttachedToCP(
 	return cert
 }
 
+// KonnectExtension deploys a KonnectExtension.
+func KonnectExtension(
+	t *testing.T,
+	ctx context.Context,
+	cl client.Client,
+	opts ...ObjOption,
+) *konnectv1alpha1.KonnectExtension {
+	t.Helper()
+
+	ke := &konnectv1alpha1.KonnectExtension{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "konnect-extension-",
+		},
+	}
+	for _, opt := range opts {
+		opt(ke)
+	}
+
+	require.NoError(t, cl.Create(ctx, ke))
+	logObjectCreate(t, ke)
+
+	return ke
+}
+
+// KonnectExtensionAttachedToCP deploys a KonnectExtension attached to a Konnect CP represented by the given KonnectGatewayControlPlane.
+func KonnectExtensionAttachedToCP(
```

Comment:
```suggestion
func KonnectExtensionRefencingKonnectGatewayControlPlane(
```

The word attachedToCP is misleading as we are going to have extensibility for operator's `ControlPlane`s as well.